// ng2-sticky-nav/__ivy_ngcc__/dist/index.js
"use strict";
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('./src/sticky-nav.directive');
var ɵngcc2 = require('@angular/common');
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StickyNavModule = void 0;
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var sticky_nav_directive_1 = require("./src/sticky-nav.directive");
__exportStar(require("./src/sticky-nav.directive"), exports);
var StickyNavModule = /** @class */ (function () {
    function StickyNavModule() {
    }
    StickyNavModule_1 = StickyNavModule;
    StickyNavModule.forRoot = function () {
        return {
            ngModule: StickyNavModule_1
        };
    };
    var StickyNavModule_1;
StickyNavModule.ɵfac = function StickyNavModule_Factory(t) { return new (t || StickyNavModule)(); };
StickyNavModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: StickyNavModule });
StickyNavModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [common_1.CommonModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(StickyNavModule, [{
        type: core_1.NgModule,
        args: [{
                imports: [
                    common_1.CommonModule
                ],
                declarations: [
                    sticky_nav_directive_1.StickyNavDirective
                ],
                exports: [
                    sticky_nav_directive_1.StickyNavDirective
                ]
            }]
    }], function () { return []; }, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(StickyNavModule, { declarations: [ɵngcc1.StickyNavDirective], imports: [ɵngcc2.CommonModule], exports: [ɵngcc1.StickyNavDirective] }); })();
    return StickyNavModule;
}());
exports.StickyNavModule = StickyNavModule;
