// ngx-scrolltop/esm2020/lib/ngx-scrolltop.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgxScrollTopComponent } from './ngx-scrolltop.component';
import { NgxScrollTopDirective } from './ngx-scrolltop.directive';
import { NgxScrollTopCoreService } from './ngx-scrolltop.core.service';
import * as i0 from "@angular/core";
export class NgxScrollTopModule {
}
NgxScrollTopModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.2.3", ngImport: i0, type: NgxScrollTopModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxScrollTopModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.2.3", ngImport: i0, type: NgxScrollTopModule, declarations: [NgxScrollTopComponent, NgxScrollTopDirective], imports: [CommonModule], exports: [NgxScrollTopComponent, NgxScrollTopDirective] });
NgxScrollTopModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.2.3", ngImport: i0, type: NgxScrollTopModule, providers: [NgxScrollTopCoreService], imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.2.3", ngImport: i0, type: NgxScrollTopModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgxScrollTopComponent, NgxScrollTopDirective],
                    imports: [CommonModule],
                    providers: [NgxScrollTopCoreService],
                    exports: [NgxScrollTopComponent, NgxScrollTopDirective],
                }]
        }] });
